core/src/main/java/io/questdb/griffin/engine/functions/regex/AbstractLikeStrFunctionFactory.java

```diff
@@ -195,10 +203,18 @@ public BindLikeStrFunction(Function value, Function pattern, boolean caseInsensi
 
         @Override
         public boolean getBool(Record rec) {
-            if (matcher != null) {
+            if (useEquality) {
                 CharSequence cs = value.getStrA(rec);
+                if (cs == null) return false;
+                if (caseInsensitive) return cs.toString().toLowerCase().equals(exactPattern);
```

Comment:
there is opportunity to do something really creative here. toString() and toLowercase() on critical path is a no-go unfortunately.

Consider isolating these cases:
- ASCII only comparison (lowecasing ASCII is a simple a | 32)
- consider using SWAR approach (instead of byte-by-byte comparisor)